client/bank/lcd/query.go

```diff
@@ -74,34 +74,90 @@ func QueryCoinTypeRequestHandlerFn(cdc *codec.Codec, cliCtx context.CLIContext)
 }
 
 // QueryTokenStatsRequestHandlerFn performs token statistic query
-func QueryTokenStatsRequestHandlerFn(cdc *codec.Codec, cliCtx context.CLIContext) http.HandlerFunc {
+func QueryTokenStatsRequestHandlerFn(cdc *codec.Codec, decoder auth.AccountDecoder, cliCtx context.CLIContext) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		resToken, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/%s", protocol.AccountRoute, bankv1.QueryTokenStats), nil)
-		if err != nil {
-			utils.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
-		}
+		vars := mux.Vars(r)
+		assetId := vars["id"]
 
-		var tokenStats bank.TokenStats
-		err = cdc.UnmarshalJSON(resToken, &tokenStats)
-		if err != nil {
-			utils.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
-			return
-		}
+		if len(assetId) == 0 || isIris(assetId) {
```

Comment:
`id` is a required param of `/bank/token-stats/{id}`, it can not be routed here if the `id` is empty.